File: C++/28.cpp
#include <iostream>
#include <cmath>

using namespace std;

double Discriminant(double a, double b, double c)
{
    double d = ((b * b) - (4 * a * c));

    return d;
}

int main()
{
    int a, b, c;
    cin >> a >> b >> c;

    double d = Discriminant(a, b, c);

    if (d > 0)
    {
        double x1 = (-b + sqrt(d)) / (2 * a);
        double x2 = (-b - sqrt(d)) / (2 * a);

        cout << "The solutions of the equation are: " << endl;
        cout << x1 << " " << x2;
    }

    else if (d == 0)
    {
        double x = -b / (2 * a);
        cout << "The solution of the equation is: " << endl;
        cout << x;
    }

    else
    {
        cout << "There are no real solutions for this equation";
    }
}

File: Python/28.py
import math

def solve_quadratic(a, b, c):
    discriminant = b**2 - 4*a*c 

    if discriminant > 0:
        x1 = (-b + math.sqrt(discriminant)) / (2 * a)
        x2 = (-b - math.sqrt(discriminant)) / (2 * a)
        return (x1, x2)
    elif discriminant == 0:
        x = -b / (2 * a)
        return (x,)
    else:
        return None

a = float(input("Введите a: "))
b = float(input("Введите b: "))
c = float(input("Введите c: "))


if a == 0:
    print("не квадратное уравнение, так как a = 0.")
else:
    roots = solve_quadratic(a, b, c)

    if roots is None:
        print("не имеет действительных решений.")
    elif len(roots) == 1:
        print(f"имеет один корень: {roots[0]}")
    else:
        print(f"имеет два корня: {roots[0]} и {roots[1]}")
